# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/events/EventStateManager.cpp
# Commit: 0ea945553f63
# Full Hash: 0ea945553f63659b99fd37265c5e7a764f70504d
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-12-03 16:21:08
# Regressor Bug: 1680669
# File Overlap Count: 1
# Description:
#   Bug 1680669 - part 6: Make methods of `MouseEvent` and static methods of `Event` aware of fractional coordinates r=smaug,emilio,webdriver-reviewers
#   
#   `screenX`, `screenY`, `clientX`, `clientY`, `x`, `y`, `offsetX` and `offsetY`
#   are now `double`.  This patch makes the event classes aware of DOM events.
#   
# ==============================================================================

diff -r 98d5a05c6c6e -r 0ea945553f63 dom/events/EventStateManager.cpp
--- a/dom/events/EventStateManager.cpp	Tue Dec 03 07:10:01 2024 +0000
+++ b/dom/events/EventStateManager.cpp	Tue Dec 03 07:10:05 2024 +0000
@@ -942,11 +942,11 @@
     // XXX Probably doesn't matter much, but storing these in CSS pixels instead
     // of device pixels means behavior can be a bit odd if you zoom while
     // pointer-locked.
-    sLastScreenPoint =
+    sLastScreenPoint = RoundedToInt(
         Event::GetScreenCoords(aPresContext, aEvent, aEvent->mRefPoint)
-            .extract();
-    sLastClientPoint = Event::GetClientCoords(
-        aPresContext, aEvent, aEvent->mRefPoint, CSSIntPoint(0, 0));
+            .extract());
+    sLastClientPoint = RoundedToInt(Event::GetClientCoords(
+        aPresContext, aEvent, aEvent->mRefPoint, CSSDoublePoint{0, 0}));
   }
 
   *aStatus = nsEventStatus_eIgnore;
@@ -4420,9 +4420,9 @@
           // to set drag end point in such case (you hit assersion if you do
           // it).
           if (sourceWC) {
-            CSSIntPoint dropPointInScreen =
+            const CSSIntPoint dropPointInScreen = RoundedToInt(
                 Event::GetScreenCoords(aPresContext, aEvent, aEvent->mRefPoint)
-                    .extract();
+                    .extract());
             dragSession->SetDragEndPointForTests(dropPointInScreen.x,
                                                  dropPointInScreen.y);
           }